benchmarks/bench_pickle.py

```diff
@@ -0,0 +1,455 @@
+"""
+Benching joblib pickle I/O.
+
+Warning: this is slow, and the benchs are easily offset by other disk
+activity.
+"""
+import os
+import time
+import shutil
+import numpy as np
+import joblib
+import gc
+
+from joblib.disk import disk_used
+
+try:
+    from memory_profiler import memory_usage
+except ImportError:
+    memory_usage = None
+
+
+def clear_out():
+    """Clear output directory."""
+    if os.path.exists('out'):
+        shutil.rmtree('out')
+    os.mkdir('out')
+
+
+def kill_disk_cache():
+    """Clear disk cache to avoid side effects."""
+    if os.name == 'posix' and os.uname()[0] == 'Linux':
+        try:
+            os.system('sudo sh -c "sync; echo 3 > /proc/sys/vm/drop_caches"')
+        except IOError as e:
+            if e.errno == 13:
+                print('Please run me as root')
+            else:
+                raise e
```

Comment:
You should use `raise` without any argument to reraise the same exception. Same as `throw` in C++ IIRC.